src/index/StringSortComparator.h

```diff
@@ -51,11 +51,21 @@ class LocaleManager {
     [[nodiscard]] const std::string& get() const { return _content; }
     std::string& get() { return _content; }
 
-    // compare according to the byte value
+    // Comparison of sort key is done lexicographically on the byte values
+    // of member `_content`
     [[nodiscard]] int compare(const SortKey& rhs) const {
       return _content.compare(rhs._content);
     }
 
+    auto operator<=>(const SortKey&) const = default;
+    bool operator==(const SortKey&) const = default;
+
+    /// Is this sort key a prefix of another sort key. Note: This does not imply
+    /// any guarantees on the relation of the underlying strings.
+    bool starts_with(const SortKey& rhs) {
+      return get().starts_with(rhs.get());
+    }
+
```

Comment:
If `get` is marked as `constexpr`, this function can be too if I'm not missing something obvious?

Reply:
The obvious thing you are missing is,
that constexpr string  is not yet implemented in GCC/clang and then also
the `starts_with` function of `string` is not constexpr (which would also be pointless).
I have added a TODO, there is quite some other stuff to do.

But, what you could have said, is that `starts_with` is `const noexcept` and
some other functions of this class are also noexcept.

Reply:
Of course, I only checked the string_view variant, which is obviously constexpr